File: src/main/js/entities/Sprint.ts
import { Record } from 'immutable';
import { defaultProps as issuesContainerDefaultProps, IssuesContainerProps } from './IssuesContainer';

export enum SprintStatus {
  PLANNED= 'PLANNED',
  PENDING= 'PENDING',
  COMPLETED= 'COMPLETED'
}

export interface SprintProps extends IssuesContainerProps {
  name: string,
  goal: string,
  startDate: string,
  endDate: string
  type: 'Sprint'
  status: SprintStatus
}

const defaultSprintProps: SprintProps = {
  ...issuesContainerDefaultProps,
  name: '',
  goal: '',
  startDate: '',
  endDate: '',
  type: 'Sprint',
  status: SprintStatus.PLANNED,
};

export default class Sprint extends Record(defaultSprintProps) {}

File: src/main/java/issuetracker/models/Sprint.java
package issuetracker.models;

import issuetracker.models.common.IssueContainer;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

import javax.persistence.Entity;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.validation.constraints.NotBlank;
import javax.validation.constraints.NotNull;
import java.util.Date;

import static issuetracker.models.Sprint.SprintStatus.PLANNED;

@Entity
@Getter
@Setter
@ToString
@NoArgsConstructor
public class Sprint extends IssueContainer {
    @ManyToOne
    @JoinColumn(updatable = false)
    @NotNull
    private Project project;

    @NotBlank(message = "Name must not be empty")
    private String name;

    private String goal;

    private Date startDate;

    private Date endDate;

    private SprintStatus status = PLANNED;

    public enum SprintStatus {
        PLANNED,
        PENDING,
        COMPLETED
    }
}
